### PR TITLE
KS-AMC-F-05 echo protocol

### PR DESCRIPTION
**Pull request by markya0616**
**Before submitting a pull request**, please make sure the following is done:

### Checklist

- [ ] Fork [the repository](https://github.com/getamis/alice) and create your new branch from `master`.
- [ ] Please mention the name of community in Pull Request title.
- [ ] Git Commit Messages
    - [ ] Use the present tense (Also in Pull Request title): "Add feature" not "Added feature"
    - [ ] Use the imperative mood (Also in Pull Request title): "Move cursor to..." not "Moves cursor to..."
    - [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.

### CODE REVIEW

File: crypto/tss/ecdsa/cggmp/sign/message.go [hunk 1]
```diff
@@ -42,14 +40,21 @@ func (m *Message) GetMessageType() types.MessageType {
...
-	got := blake2b.Sum256(bs)
-	return got[:], nil
+	switch m.Type {
+	case Type_Round1:
+		mm.Body = &Message_Round1{
+			Round1: &Round1Msg{
+				KCiphertext:     m.GetRound1().GetKCiphertext(),
+				GammaCiphertext: m.GetRound1().GetGammaCiphertext(),
+				// Not broadcast to all in echo protocol
+				// Psi:             m.GetRound1().GetPsi(),
```
Comment (cychuang0924) [inline crypto/tss/ecdsa/cggmp/sign/message.go:55]: remove psi ?

File: crypto/tss/ecdsa/cggmp/signSix/message.go [hunk 2]
```diff
@@ -48,14 +46,23 @@ func (m *Message) GetMessageType() types.MessageType {
...
+	switch m.Type {
+	case Type_Round1:
+		mm.Body = &Message_Round1{
+			Round1: &Round1Msg{
+				KCiphertext:     m.GetRound1().GetKCiphertext(),
+				GammaCiphertext: m.GetRound1().GetGammaCiphertext(),
+				Z1:              m.GetRound1().GetZ1(),
+				Z2:              m.GetRound1().GetZ2(),
+				// Not broadcast to all in echo protocol
+				// Psi:             m.GetRound1().GetPsi(),
```
Comment (cychuang0924) [inline crypto/tss/ecdsa/cggmp/signSix/message.go:63]: The same here ?

Comment (markya0616) [reply crypto/tss/ecdsa/cggmp/signSix/message.go:63]: just a reminder indicating we don't send out Psi in echo message

File: crypto/tss/ecdsa/cggmp/sign/message.go (hunk 1 above)
Comment (markya0616) [reply crypto/tss/ecdsa/cggmp/sign/message.go:55]: ditto

**Review by cychuang0924 — APPROVED** [review]

**Pull request merged**